src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -746,6 +746,9 @@ inline COOToCSRAlg WhichCOOToCSR(const COOMatrix &coo) {
 
 template <DGLDeviceType XPU, typename IdType>
 CSRMatrix COOToCSR(COOMatrix coo) {
+  if constexpr (std::is_same<IdType, int32_t>::value) {
```

Comment:
why check for `int32_t` only?

Reply:
Do you mean that we need to check whether `NNZ` exceeds `INT64_MAX` when `IdType` is `int64_t`? That's a huge number and I don't think anyone would have the requirement to handle such a huge COO in 2024.

BTW, currently `NNZ` has fixed type `int64_t`. See https://github.com/dmlc/dgl/blob/ed50c170dda9627730cb8ee4c7110205b6ea09de/include/dgl/runtime/c_runtime_api.h#L188

Reply:
in theory, we'd better always check the overflow for any `IdType` and `NNZ`. But For now, only int32_t and int64_t are available for `IdType`. So we check only for int32_t is sufficient as it's not necessary to check for int64_t.
But without the `if` looks more straightforward, robust and clearer. It will not cause issue if we support more `IdType`s in the future.